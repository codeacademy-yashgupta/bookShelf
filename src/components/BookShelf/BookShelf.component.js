import React, { Component } from 'react';
import './BookShelf.css';
import GroupBookCards from '../GroupBookCards/GroupBookCards.component';
import mockData from '../../mockData.json';
import { httpGet } from '../../helpers/httpGet';

class BookShelf extends Component {
  state= {
    booksData: [],
  }

  APIForGettingBooksWithRating = 'http://localhost:3001/booksWithRating';

  componentDidMount() {
    httpGet(this.APIForGettingBooksWithRating)
      .then((response) => {
        this.setState({ booksData: response.data });
      });
  }

  render() {
    const { booksData } = this.state;
    const authorsArray = Object.keys(booksData);
    const rows = [];
    authorsArray.forEach((author) => {
      rows.push(<GroupBookCards author={author} authorRows={booksData[author]} />);
    });
    return (
      <div className="main-body">
        {rows}
      </div>
    );
  }
}

export default BookShelf;
