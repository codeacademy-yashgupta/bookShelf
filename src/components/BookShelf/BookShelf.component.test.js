import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import BookShelf from './BookShelf.component';

configure({ adapter: new Adapter() });
describe('BookShelf', () => {
  it('should render book shelf', () => {
    const tree = renderer.create(<BookShelf />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('componentDidMount should be called', () => {
    const componentDidMountSpy = jest.spyOn(BookShelf.prototype, 'componentDidMount');
    const wrapper = shallow(<BookShelf />);
    expect(BookShelf.prototype.componentDidMount).toHaveBeenCalled();
  });
});
