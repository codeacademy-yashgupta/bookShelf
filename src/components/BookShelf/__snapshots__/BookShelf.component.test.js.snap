// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookShelf should render book shelf 1`] = `
<div
  className="main-body"
/>
`;
